x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -216,4 +219,22 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       projectsUrl: fullCloudProjectsUrl,
     };
   }
+
+  private async fetchElasticsearchConfig(
+    http: CoreStart['http']
+  ): Promise<PublicElasticsearchConfigType> {
+    if (this.elasticsearchUrl) {
+      return { elasticsearchUrl: this.elasticsearchUrl };
+    }
+    try {
+      const result = await http.get<ElasticsearchConfigType>('/api/internal/elasticsearch_config');
+      this.elasticsearchUrl = result.elasticsearch_url;
```

Comment:
Should this treat `elasticsearch_url === ''` as `undefined`? 

Granted we'd need to handle the caching better in that case because we wouldn't want to refetch every time if the server doesn't have a value.

It looks like all the usages we're updated to use `||` instead of `??` so this should be mostly handled correctly now. But I do worry about a consumer understanding the nuance of this type.

Reply:
I changed it to save a config object, and always return undefined for ''. That way we can do caching (should only ever fetch once) and be unambiguous.